src/index.js

```diff
@@ -52,12 +53,12 @@ function loadConfig(configPath) {
  * @returns {Promise<boolean>} Promise of whether the linting passed or failed
  */
 module.exports = function lintStaged(
-  { configPath, relative = false, shell = false, quiet = false, debug = false } = {},
+  { configPath, config, relative = false, shell = false, quiet = false, debug = false } = {},
   logger = console
 ) {
   debugLog('Loading config using `cosmiconfig`')
 
-  return loadConfig(configPath)
+  return (config ? Promise.resolve(config) : loadConfig(configPath))
```

Comment:
Can you wrap the config object inside `validateConfig`, so that it still gets validated in this case? It will return the (same) config if it's valid and throw errors otherwise.

Reply:
I made fix so config is resolved in the same format as loadConfig and because of this it is correctly passed to validateConfig further down the file